devito/operator/operator.py

```diff
@@ -1388,6 +1389,38 @@ def _get_nbytes(self, i):
 
         return nbytes
 
+    @cached_property
+    def _visible_devices(self):
+        device_vars = (
+            'CUDA_VISIBLE_DEVICES',
+            'ROCR_VISIBLE_DEVICES',
+            'HIP_VISIBLE_DEVICES'
+        )
+        for v in device_vars:
+            try:
+                return tuple(int(i) for i in os.environ[v].split(','))
+            except (ValueError, KeyError):
```

Comment:
Should these be split?
- ValueError -> there is an id so set it to `os.environ[v].split(',').index(i)`
- KeyError -> no env var, no device